spec/controllers/your_details_controller_spec.rb

```diff
@@ -100,6 +102,63 @@ module Wpcc
       end
     end
 
+    describe 'sessions' do
+      after { Timecop.return }
+
+      it "'wpcc_expires_at' key should always be updated" do
+        session[:wpcc_expires_at] = 5.minutes.ago
+        get :new, nil, some_session
+
+        expect(session[:wpcc_expires_at]).to be > 5.minutes.ago
+      end
+
+      context 'when session has expired' do
+        before do
+          wpcc_expires_at = 5.minutes.ago
+
+          Timecop.travel(Time.current + 30.minutes)
+
+          get :new,
+              nil,
+              age: 34,
+              gender: 'male',
+              salary: 9000,
+              salary_frequency: 'yearly',
+              contribution_preference: 'full',
+              wpcc_expires_at: wpcc_expires_at
+        end
+
+        it 'resets the wpcc specific session data to nil' do
+          expect(session[:age]).to eq nil
+          expect(session[:gender]).to eq nil
+          expect(session[:salary]).to eq nil
+          expect(session[:salary_frequency]).to eq nil
+          expect(session[:contribution_preference]).to eq nil
+        end
+      end
+
+      context 'session has NOT expired' do
+        before do
+          get :new,
+              nil,
+              age: 34,
```

Comment:
again.
For this can we not use some_session hash with the wpcc_expires_at variable pushed into it?
Rather than creating a whole new set of data here